PGM/src/main/java/tc/oc/pgm/polls/types/PollKick.java

```diff
@@ -21,33 +21,32 @@
         PollKick create(PlayerId initiator, Player player);
     }
 
-    private final Player player;
+    private final PlayerId player;
     private final IdentityProvider identityProvider;
     private final PunishmentCreator punishmentCreator;
     private final UserStore userStore;
 
     @AssistedInject PollKick(@Assisted PlayerId initiator, @Assisted Player player, PollManager pollManager, Audiences audiences, IdentityProvider identityProvider, PunishmentCreator punishmentCreator, UserStore userStore) {
         super(pollManager, initiator, audiences);
-        this.player = player;
         this.identityProvider = identityProvider;
         this.punishmentCreator = punishmentCreator;
         this.userStore = userStore;
+        this.player = userStore.playerId(player);
     }
 
     @Override
     public void executeAction() {
-        PlayerId kicked = userStore.playerId(player);
-        punishmentCreator.create(null, kicked, "The poll to kick " + kicked.username() + " succeeded", PunishmentDoc.Type.KICK, null, false, false, true);
+        punishmentCreator.create(null, player, "The poll to kick " + player.username() + " succeeded", PunishmentDoc.Type.KICK, null, false, false, true);
```

Comment:
Maybe make this translatable instead of a hardcoded string

Reply:
Well, It doesn't really make sense to translate a punishment message, as it cannot be translated per user.

Reply:
what about translations?